handler/transport.go

```diff
@@ -42,10 +47,48 @@ func (t *transport) RoundTrip(r *http.Request) (*http.Response, error) {
 		}
 	}
 
-	if err := t.prov.Verify(r, r.Method, r.URL.Path); err != nil {
+	reqBodyClosed := false
+	if r.Body != nil {
+		defer func() {
+			if !reqBodyClosed {
+				r.Body.Close()
+			}
+		}()
```

Comment:
Please also add unit test for this block. I think we can reference to the original code.
https://github.com/golang/oauth2/blob/bf48bf16ab8d622ce64ec6ce98d2c98f916b6303/transport_test.go

Reply:
Fixed
https://github.com/yahoojapan/authorization-proxy/pull/66/commits/93d1112f058a87d924c1ef7b46faeabe30e1ef6d